packages/remark-stringify/types/index.d.ts

```diff
@@ -0,0 +1,14 @@
+// TypeScript Version: 3.0
+
+declare module 'remark-stringify' {
+  import {RemarkStringifyOptions} from 'remark-stringify/types'
```

Comment:
Let's keep everything in one file.

Reply:
Happy to co-locate everything to one source file if that's your preference.

Are you on board with using a "ghost module" as a safe way to expose type information (consistent with [DefinitelyTyped](http://definitelytyped.org/guides/best-practices.html) best practices). This would take the form of a single `.ts` file like

```ts
declare module 'remark-stringify/types' {
  ...
}

declare module 'remark-stringify' {
  import {RemarkStringifyOptions} from 'remark-stringify/types'

  ...
}

```

Reply:
The document looks quite old.

As long as I know the recommendation example,

```ts
declare module X
```

, now becomes

```ts
declare namespace X
```
> https://www.typescriptlang.org/docs/handbook/modules.html

Reply:
Let's keep `RemarkStringifyOptions` in `remark-stringify` rather than in  `remark-stringify/types`. I cannot understand why do you want to keep it in the different path.

Reply:
>  I cannot understand why do you want to keep it in the different path.

The benefit of a "ghost module" is you never have the potential to collide with current (or future) value exports from the `remark-stringify` module.

I'm happy to align with the maintainers' preferences, but if I were writing my own ambient types for my own library, I'd look for alignment with the standard DefinitelyTyped style

Reply:
I've tried this a couple of ways, and can't figure out how to get a commonJS style export (`export = foo`) along with a namespace that contains the options.

At this point, I'll have ask you to propose a concrete approach that you can verify works w/ the current (or similar) tests

Reply:
> The benefit of a "ghost module" is you never have the potential to collide with current (or future) value exports from the remark-stringify module.

It would be nice to have, but contributors also need to know where other interfaces exist.

> At this point, I'll have ask you to propose a concrete approach that you can verify works w/ the current (or similar) tests

 I'll make another pr to your pr. 👍

Reply:
Btw, thanks for your patient.